gpiozero/other_devices.py

```diff
@@ -147,6 +147,92 @@ def is_active(self):
         """
         return self.temperature > self.threshold
 
+class LoadAverage(InternalDevice):
+    """
+    Extends :class:`InternalDevice` to provide a device which is active when
+    the CPU load average exceeds the *threshold* value.
+
+    The following example plots the load average on an LED bar graph::
+
+        from gpiozero import LEDBarGraph, LoadAverage
+        from signal import pause
+
+        la = LoadAverage(min_load_average=0, max_load_average=2)
+        graph = LEDBarGraph(5, 6, 13, 19, 25, pwm=True)
+
+        graph.source = la.values
+
+        pause()
+
+    :param str load_average_file:
+        The file from which to read the load average. This defaults to the
+        proc file :file:`/proc/loadavg`. Whatever file is specified is expected
+        to contain three space-separated load averages at the beginning of the
+        file, representing 1 minute, 5 minute and 15 minute averages
+        respectively.
+
+    :param float min_load_average:
+        The load average at which :attr:`value` will read 0.0. This defaults to
+        0.0.
+
+    :param float max_load_average:
+        The load average at which :attr:`value` will read 1.0. This defaults to
+        1.0.
+
+    :param float threshold:
+        The load average above which the device will be considered "active".
+        This defaults to 0.8.
+
+    :param int minutes:
+        The number of minutes over which to average the load. Must be 1, 5 or
+        15.
+    """
+    def __init__(self, load_average_file='/proc/loadavg', min_load_average=0.0,
+        max_load_average=1.0, threshold=0.8, minutes=1):
+        self.load_average_file = load_average_file
+        super(InternalDevice, self).__init__()
+        self.min_load_average = min_load_average
+        self.max_load_average = max_load_average
+        self.threshold = threshold
+        if minutes not in (1, 5, 15):
+            raise ValueError('minutes must be 1, 5 or 15')
+        self._load_average_file_column = {
+            1: 0,
+            5: 1,
+            15: 2,
+        }[minutes]
```

Comment:
Ah, Python 3's chained exceptions. Personally I'd still prefer the try/except version (EAFP vs LBYL and all that; not that it matters in this case but the style is just ingrained in me!). It does make the error longer but I wouldn't say *much* scarier. Someone who doesn't understand/read the exceptions will be scared off in either case by all the preamble, and someone who does know to read the last line first gets the message straight away in either case. Incidentally, if you're *really* bothered about the extra chained exception you can just do:

```python
try:
    self._load_average_file_column = { ... }[minutes]
except KeyError:
    raise ValueError('minutes must be 1, 5 or 15') from None
```

The `from None` squishes the exception chain (and is compatible with Python 2.x).